apps/server/src/modules/files-storage-client/service/copy-files.service.spec.ts

```diff
@@ -52,16 +53,16 @@ describe('copy files service', () => {
 	describe('copy files of entity', () => {
 		const setup = () => {
 			const school = schoolFactory.build();
-			const file1 = fileFactory.buildWithId({ name: 'file.jpg' });
-			const file2 = fileFactory.buildWithId({ name: 'file.jpg' });
+			const file1 = { id: new ObjectId().toHexString(), name: 'file.jpg' };
+			const file2 = { id: new ObjectId().toHexString(), name: 'file.jpg' };
```

Comment:
Please use fileRecordFactory

Reply:
You mean fileFactory or? The fileFactory match to the legacy code stuff.

Reply:
@SevenWaysDP I think you meant the _fileFactory_ (the legacy files storage one, not the new _fileRecordFactory_ as @CeEv pointed out).

But the question is: why? The code that use these objects doesn't care if the object is of the _FileEntity_ class. I've changed that on purpose to decouple the _files-storage-client_ module from the _files_ module - thanks to this change we no longer have any external coupling to the files module entities which are now hidden inside of it and not accessible from the outside world (same as the factory for the entities) - first, I think it's a good practice to do that whenever possible (and just rely on some abstract interface) and second, thanks to this it would be easier to create a separate microservice out of it if there will be a decision to do so.

Reply:
@SevenWaysDP please look at my comment above - your change request will block the PR and I think my reasons not to do that are valid, let me know what you think.

Reply:
I would say go with a local "legacyFileMockFactory" or something

Reply:
@Metauriel  Ok I'll add some kind of a factory for these calls, it feels a bit like an overkill as the factory code will be bigger than these two calls but if this will conform the overall tests convention across the project I think it's worth it :blush: . Also I'm happy we've agreed to get rid of the direct dependency to the legacy _FileEntity_.

Reply:
Had to move it to the `@shared/testing` ( https://github.com/hpi-schul-cloud/schulcloud-server/pull/4303/commits/6d71fae93a88d4ab05deabd4533006bf2d67bbea ) as it's also needed in the `task-copy.service.spec.ts` as pointed out by Sergej in https://github.com/hpi-schul-cloud/schulcloud-server/pull/4303#discussion_r1305494413 .